dashboard/modules/job/job_head.py

```diff
@@ -31,6 +39,52 @@ def __init__(self, dashboard_head):
         # JobInfoGcsServiceStub
         self._gcs_job_info_stub = None
 
+    async def _next_job_id(self):
+        counter_str = await self._dashboard_head.aioredis_client.incr(
+            job_consts.REDIS_KEY_JOB_COUNTER)
+        job_id_int = int(counter_str)
+        return ray.JobID.from_int(job_id_int)
```

Comment:
Oh this looks really problematic... we definitely shouldn't be talking directly to redis here and this is also introducing a race condition if another driver is started before we actually start the job.

Reply:
`incr` is an atomic increment operation in redis. Every job uses `incr` to get a new job id, so there is no race condition. Please refer to: https://redis.io/commands/INCR

Reply:
Ok, that's fine but we should definitely not talk directly to redis. If we absolutely need this it should go through the GCS service. Don't want to be adding new dependencies on redis when we're trying to remove it as a hard dependency.

Reply:
How to get the job status if we got failures when starting the job? The driver may be started or failed when the connection between dashboard and job agent is lost. If we have a job id, then we have a key to query the job status or drop the job.

Our original design is to submit the job to the GCS from dashboard.